src/main/java/org/snomed/snowstorm/core/data/services/ConceptUpdateHelper.java

```diff
@@ -133,10 +137,17 @@ PersistedComponents saveNewOrUpdatedConcepts(Collection<Concept> concepts, Commi
 			// Mark changed concepts as changed
 			if (existingConcept != null) {
 				concept.setCreating(false);// May have been set true earlier during first save
+				
+				//Are we copying the released details from the existing concept, or has the source been versioned recently?
+				Concept authoratativeConcept = existingConcept;
```

Comment:
It would be cleaner to do this within a new version of the `SnomedComponent.copyReleaseDetails` method that takes both the existingConcept and existingSourceConcept. The same should apply for the other component types too!
Also I think this logic should take the component with the greatest 'getReleasedEffectiveTime()'.

Reply:
OK, yes of course I should have done this for all component types.    I've created public void copyReleaseDetails(SnomedComponent<C> component, SnomedComponent<C> existingRebaseSourceComponent)  and then I commented out the previous one temporarily to see all the places it was being called and switched over to the rebase version where possible. 

The only place I wasn't sure about (because it gets called from A LOT) of places is ReferenceSetMemberService @ 374 which I suspect needs to have an optional rebaseBranch passed in so we can call our new function at line 380 if we have more than one existing member to consider.